tempto-core/src/main/java/com/teradata/tempto/internal/fulfillment/table/hive/HiveTableManager.java

```diff
@@ -129,31 +156,14 @@ else if (state == LOADED) {
             uploadTableData(tableDataPath, tableDefinition.getDataSource());
         }
 
-        if (state == LOADED && analyzeMutableTables) {
-            Optional<List<HiveTableDefinition.PartitionDefinition>> partitionDefinitons = Optional.empty();
-            if (tableDefinition.isPartitioned()) {
-                partitionDefinitons = Optional.of(tableDefinition.getPartitionDefinitons());
-            }
-            analyzeTable(tableName, partitionDefinitons);
+        if (state == LOADED && analyzeMutableTables && tableDefinition.getDataSource().getStatistics().isPresent()) {
+            checkState(!tableDefinition.isPartitioned(), "Statisitcs are not yet supported for parititioned tables");
+            hiveThriftClient.setStatistics(tableName, tableDefinition.getDataSource().getStatistics().get());
         }
 
         return new HiveTableInstance(tableName, tableDefinition);
     }
 
-    private void analyzeTable(TableName tableName, Optional<List<HiveTableDefinition.PartitionDefinition>> partitionDefinitions)
```

Comment:
scrapping old functionality could be done in separate commit. I will not force push too much though.